scaleset_resources.tf

```diff
@@ -100,6 +100,20 @@ resource "azurerm_network_security_rule" "security_rule_ssh" {
   network_security_group_name = azurerm_network_security_group.nsg.name
 }
 
+resource "azurerm_network_security_rule" "security_rule_metrics" {
```

Comment:
This resource should be conditional if `var.metrics_source_address_prefixes` is actually provided.